src/Compilers/CSharp/Portable/Symbols/Retargeting/RetargetingMethodSymbol.cs

```diff
@@ -280,6 +280,19 @@ private ImmutableArray<MethodSymbol> RetargetExplicitInterfaceImplementations()
             return builder.ToImmutableAndFree();
         }
 
+        /// <summary>
+        /// Retarget the underlying method's overridden method.
+        /// </summary>
+        internal MethodSymbol RetargetOverriddenMethod
+        {
+            get
+            {
+                return _underlyingMethod.OverriddenMethod is MethodSymbol overridden
+                    ? this.RetargetingTranslator.Retarget(overridden, MemberSignatureComparer.RetargetedExplicitImplementationComparer)
```

Comment:
>this.RetargetingTranslator.Retarget(overridden, MemberSignatureComparer.RetargetedExplicitImplementationComparer) [](start = 22, length = 113)

Retargeting the overridden method calculated in a different "universe" feels like a wrong thing to do. I think we should re-resolve overriding instead, otherwise we will not match result with metadata when an override is added/removed, etc. in the new compilation.

Reply:
This method is the overriding method. I presume you mean re-resolve the overridden method? That is indeed what is done in `OverriddenOrHiddenMembersHelpers`. However, the value returned by this property is used in the face of covariant returns, just as it is will be used by the runtime in the face of covariant returns unless the retargeted method is recompiled with the revised reference. In other words, the intent of this property is to assist `OverriddenOrHiddenMembersHelpers` in modeling what the runtime does rather than modeling what would happen if the retargeted method were recompiled.

If you still think this is incorrect, can you please be more explicit about the scenario that you believe might behave incorrectly?

---
In reply to: [441831407](https://github.com/dotnet/roslyn/pull/44025#discussion_r441831407) [](ancestors = 441831407)

Reply:
> However, the value returned by this property is used in the face of covariant returns

I see no indication in this method (neither in its name, or in its implementation) that we are dealing with an override that will have an explicit MethodImpl emitted in metadata. Only under these conditions the current implementation would be correct, I think. If that is the only scenario when this method is supposed to be used, I suggest a rename to indicate this restriction and an assert inside to verify the MethodImpl requirement.

---
In reply to: [441896761](https://github.com/dotnet/roslyn/pull/44025#discussion_r441896761) [](ancestors = 441896761,441831407)

Reply:
That is the scenario affected by this property, but it is not necessary (or intended) to restrict its use to that or any particular scenario in the implementation for it to have the correct impact on the compiler behavior. I believe the implementation is correct as it currently exists.

---
In reply to: [441898873](https://github.com/dotnet/roslyn/pull/44025#discussion_r441898873) [](ancestors = 441898873,441896761,441831407)

Reply:
> I believe the implementation is correct as it currently exists.

I would agree if that method were to always produce the same result as the ```OverriddenMethod``` API for scenarios when explicit MethodImpl is not involved. However, I do not believe this is the case. For example, I suspect that would not be the case when another override is injected above this method in a new compilation. Please add a direct test for this API for a scenario like that. This is not a private method, its name is very unspecific and its general use is error prone. In order for developers to not misuse this method in the future. I am asking you:
- to rename it so that it reflects the intended usage of the method
- to add an assert to verify that ```overridden``` is indeed going to have an explicit MethodImpl. 

I believe that under other conditions the result returned by this method is meaningless and shouldn't be used.  

---
In reply to: [441921131](https://github.com/dotnet/roslyn/pull/44025#discussion_r441921131) [](ancestors = 441921131,441898873,441896761,441831407)

Reply:
After reviewing more code I think we should call this method ExplicitlyOverriddenClassMethod (as in PEMethodSymbol) and implement it accordingly.

---
In reply to: [441959797](https://github.com/dotnet/roslyn/pull/44025#discussion_r441959797) [](ancestors = 441959797,441921131,441898873,441896761,441831407)